pkg/lsp/service.go

```diff
@@ -38,8 +38,20 @@ const (
 	TDSKIncremental                      = 2
 )
 
+type TextDocumentSyncOptions struct {
+	OpenClose         bool                 `json:"openClose,omitempty"`
+	Change            TextDocumentSyncKind `json:"change"`
+	WillSave          bool                 `json:"willSave,omitempty"`
+	WillSaveWaitUntil bool                 `json:"willSaveWaitUntil,omitempty"`
+	Save              *SaveOptions         `json:"save,omitempty"`
+}
+
+type SaveOptions struct {
+	IncludeText bool `json:"includeText"`
+}
+
 type ServerCapabilities struct {
-	TextDocumentSync                 int                              `json:"textDocumentSync,omitempty"`
+	TextDocumentSync                 TextDocumentSyncOptions          `json:"textDocumentSync,omitempty"`
```

Comment:
@keegancsmith To address this, I made it so that either a JSON number (TextDocumentSyncKind) or an TextDocumentSyncOptions object can be JSON-unmarshaled into a TextDocumentSyncOptions Go struct. This means we can upgrade go-langserver first at any time.